File: DigitHoles/solve/ruby/DigitHolesSolver.rb
#! /usr/bin/env ruby

require_relative "DigitHoles"

require "gettc/types"
include Gettc

def init
  gettc_home = File.expand_path(ENV["GETTC_HOME"] || File.join(ENV["HOME"], ".gettc"))
  $LOAD_PATH << File.join(gettc_home, "include", "ruby")
  require "topcoder"
  include TopCoder
end

def main
  reader = Reader.new(IO.read(ARGV[0]))
  number = reader.next(TInt)

  result = DigitHoles.new().numHoles(number)
  IO.write(ARGV[1], Writer.new.next(result, TInt).to_s)
end

init
main

File: DigitHoles/solve/cpp/DigitHolesSolver.cpp
#include <fstream>
#include <topcoder>
#include "DigitHoles.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    int number; tc::read(ifs, number);
    ifs.close();

    std::ofstream ofs(argv[2]);
    DigitHoles solver;
    tc::show(ofs, solver.numHoles(number));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}
